doc/designs/kuoros_class_diagram.md

```diff
@@ -0,0 +1,127 @@
+```mermaid
+classDiagram
+    class POSManager {
+        <<interface>>
+        createArray(CreateArrayCommand): (Response, Error)
+        createVolume (CreateVolumeCommand): (Response, Error)
+        deleteArray(DeleteArrayCommand): (Response, Error)
+        deleteVolume(DeleteVolumeCommand): (Response, Error)
+    }
+    class HAManager {
+        <<interface>>
+        listNodes(): (Response, Error)
+        listVolumes(): (Response, Error)
+        listReplication(): (Response, Error)
+        startReplication(StartReplicationCommand): (Response, Error)
+        stopReplication(StopReplicationCommand): (Response, Error)
+    }
+    class POSGRPCManager {
+        connection: GRPCConnection
+    }
+
+    class HAPostgresManager {
+        connection: PostgresConnection
+    }
+    class Response {
+        rid: string
+        lastSuccessTime: int
+        result: Result
+        info: any
+    }
+    class Result {
+        status: Status
+        data: any
+    }
+    class Status {
+        level: string
+        description: string
+        problem: string
+        solution: string
+    }
+    class Command {
+        command: string
+        rid: string
+        requestor: string
+    }
+    class CreateArrayCommand {
+        Param: CreateArrayParam
+    }
+    class CreateArrayParam {
```

Comment:
Maybe we do not need to explicitly declare the classes for Request, Response, Param, etc.?
gRPC will create them automatically?

Reply:
In my opinion, we should consider how protobuf messages will be integrated into this design.

From protobuf messages, gRPC will automatically generate many of classes, variables, and methods here.

Reply:
These classes will be the inputs and outputs of the functions that are exposed by Kuoros. We can remove it and pass/return data in a standard format like JSON.

Yes. In the current design, gRPC implementation is abstracted out. I will add that as well

Reply:
I have changed the use of classes for Commands and responses. We need to add the structure to the documentation so that users will know the exact structure